build-support/bin/check_rust_formatting.sh

```diff
@@ -34,46 +33,55 @@ while getopts "hf" opt; do
   esac
 done
 
-ensure_native_build_prerequisites >/dev/null
+NATIVE_ROOT="${REPO_ROOT}/src/rust/engine"
 
 cmd=(
-  "${CARGO_HOME}/bin/rustfmt"
-  --config-path="${NATIVE_ROOT}/rustfmt.toml"
-)
-
-files=(
-  $(find "${NATIVE_ROOT}" \
-      -name '*.rs' -not -wholename '*/bazel_protos/*' -not -wholename '*/target/*')
-  "${NATIVE_ROOT}/process_execution/bazel_protos/src/verification.rs"
-  "${NATIVE_ROOT}/process_execution/bazel_protos/build.rs"
+  ${REPO_ROOT}/build-support/bin/native/cargo fmt --all --
```

Comment:
NB: Alot of dancing here due to the switch from use of `rustfmt <files>` directly to `cargo fmt --all`. It think this was worth it, cargo has comprehension of our rust project; ie: it knows we generate proto files and skips trying to format those as you'd hope. The sticky wicket was `cargo fmt` acting on our own `src/rust/engine/process_execution/bazel_protos/src/lib.rs` which declares modules for that generated code which might not be present. More comments in `src/rust/engine/process_execution/bazel_protos/.gitignore` explain the rest of the dance along with changes in `build-support/bin/check_rust_formatting.sh` to deal with exit code 2.